src/interpolations.jl

```diff
@@ -1090,6 +1096,115 @@ function shape_value(ip::Lagrange{RefPrism, 2}, ξ::Vec{3}, i::Int)
     throw(ArgumentError("no shape function $i for interpolation $ip"))
 end
 
+
+#####################################
+# Lagrange dim 3 RefPyramid order 1 #
+#####################################
+getnbasefunctions(::Lagrange{RefPyramid,1}) = 5
+facedof_indices(::Lagrange{RefPyramid,1}) = ((1,3,4,2), (1,2,5), (1,5,3), (2,4,5), (3,5,4), )
+edgedof_indices(::Lagrange{RefPyramid,1}) = ((1,2), (1,3), (1,5), (2,4), (2,5), (4,3), (3,5), (4,5))
+ 
+function reference_coordinates(::Lagrange{RefPyramid,1})
+    return [Vec{3, Float64}((0.0, 0.0, 0.0)),
+            Vec{3, Float64}((1.0, 0.0, 0.0)),
+            Vec{3, Float64}((0.0, 1.0, 0.0)),
+            Vec{3, Float64}((1.0, 1.0, 0.0)),
+            Vec{3, Float64}((0.0, 0.0, 1.0))]
+end
+
+function shape_value(ip::Lagrange{RefPyramid,1}, ξ::Vec{3}, i::Int)
+    (x,y,z) = ξ
+    zzero = z ≈ 1.0
+    i == 1 && return zzero ? 0.0 : (-x*y+(z-1)*(-x-y-z+1))/(z-1)
```

Comment:
```suggestion
    i == 1 && return zzero ? zero(x) : (-x*y+(z-1)*(-x-y-z+1))/(z-1)
```
To make type-stable for non-Float64 input?